### ISSUE TITLE
Child windows?

### ISSUE THREAD

**Issue by cryptix** [andlabs/ui#60]
Hi,

I can't figure out how to spawn a 2nd window, I currently only see this in the OpenFile dialog and that directly touches the lower c/objc parts. Do you plan to add a more generic version?

For my use case I actually only want something like a modal messagebox to force user focus in an error case.

**Comment by cryptix** [comment]
I found a solution to emulate a MessageBox but it involves `ui.Do()` and thus I had to untangle my code so that the event handler is on another goroutine than the one which creates the main window, which might be a good idea anyway but i'd like some feedback on this.

My utility can be found [here](https://github.com/cryptix/go/blob/0159a16ac273693570335d1c7e58f45560768fb1/msgbox/msgbox.go) and [here](https://gist.github.com/cryptix/a37dd97c21087d7b990a) is some example usage.


The hiding of the parent window is debatable but It's less hassle than deactivating all the controls in it.

**Comment by andlabs** [comment]
The main reason why I have not reimplemented message boxes is because `ui.OpenFile()` still doesn't work correctly. Disabling controls will also come later.

As for child windows, I'm not entirely sure what you are asking...?

**Comment by cryptix** [comment]
> The main reason why I have not reimplemented message boxes is because ui.OpenFile() still doesn't work correctly. Disabling controls will also come later.

Aaah, I see. Okay.

> As for child windows, I'm not entirely sure what you are asking...?

Okay, the title window is missleading/wrong. The question was how to get modal dialogs / lock the parent.

I'm definitely not a fan of those in general but they fit the job quite nicely if you want to focus the user on an error that can't be recovered from.

**Comment by andlabs** [comment]
Right; modal dialogs are not in for the same reason (ui.OpenFile() not working correctly).

I'll take a look at your code again and comment on it sometime today.
